src/runtime_src/core/common/api/hw_context_int.h

```diff
@@ -28,6 +30,14 @@ get_core_device_raw(const xrt::hw_context& ctx);
 void
 set_exclusive(xrt::hw_context& ctx);
 
+// Allows the creation of the hardware context from a void pointer
+// to the hardware context implementation. This is used by the
+// XDP plugins in order to initialize when the user creates
+// a hardware context in their host
```

Comment:
Let's extend this comment to explain why the pointer is a void* rather than a forward declared xrt::hw_context_impl pointer.

Reply:
Added further explanation